bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -370,7 +370,12 @@ void ReplicaImp::tryToSendPrePrepareMsg(bool batchingLogic) {
   }
 
   pp->finishAddingRequests();
+  startConsensusProcess(pp);
```

Comment:
What is the purpose of this change? It seems unrelated to noops as far as I can tell.

Reply:
I wanted to create a single point where we send a pre prepare message and update all the counters, persist the data etc.

once we have this method we can create an empty pp message (as in viewchange) and start a consensus with it - and let the existing code to safely do its work.
Assume we in sequence number No. 140, so sending 10 such empty preprepare messages will bring the system to sequence number 150 and to the nearest checkpoint.

Reply:
Thanks for the explanation. Makes sense.